honeycomb-kernels/src/coloring/dsatur.rs

```diff
@@ -0,0 +1,95 @@
+use std::{cmp::Ordering, collections::HashMap};
+
+use honeycomb_core::{
+    cmap::{CMap2, DartIdType, Orbit2, OrbitPolicy, VertexIdType},
+    prelude::CoordsFloat,
+    stm::atomically,
+};
+use itertools::Itertools;
+use rayon::prelude::*;
+
+use super::Color;
+
+#[allow(clippy::missing_panics_doc)]
+/// DSATUR algorithm implementation
+///
+/// This algorithm is a coloring algorithm similar to the greedy coloring algorithm.
+///
+/// [Reference](https://en.wikipedia.org/wiki/DSatur).
+///
+/// # Return
+///
+/// The function will return the maximal used color, i.e. the `n_color_used - 1`.
+pub fn color<T: CoordsFloat>(cmap: &mut CMap2<T>) -> u8 {
+    cmap.add_attribute_storage::<Color>();
+
+    // build graph data as a collection of (Vertex, Vec<Neighbors>)
```

Comment:
Could this part be helpful elsewhere?
You can add a way to create a graph from a CMap. And use things like Coupe later ;).

Reply:
> Could this part be helpful elsewhere? You can add a way to create a graph from a CMap. 

Yeah, I think I can also make it generic enough to handle all possible adjacency types (e.g.  here we're using vertex-vertex, but we could fetch incident faces for another usage). 

> And use things like Coupe later ;).

I actually wanted to try and use it, but I think all coloring algorithms of coupe are focused on balancing weights, rather than making independent subsets. Is that right?